types/chai/index.d.ts

```diff
@@ -623,16 +623,15 @@ declare namespace Chai {
          * @param value   Actual value.
          * @param message   Message to display on error.
          */
-        isNotNull<T>(value: T, message?: string): void;
+        isNotNull<T>(value: T, message?: string): asserts value is (T extends null ? never : T);
 
         /**
          * Asserts that value is NaN.
          *
-         * @type T   Type of value.
          * @param value   Actual value.
          * @param message   Message to display on error.
          */
-        isNaN<T>(value: T, message?: string): void;
+        isNaN<T>(value: T, message?: string): asserts value is (T extends number ? T : never);
```

Comment:
I believe this assertion doesn't check if the value is a number, but if it is literally the value `NaN`. Since `NaN` doesn't has its own type and is just a `number`, I don't think any narrowing of `T` is possible here.